File: backend/internal/modules/chats/domain/services/chatService.go
package services

import (
	"github.com/javierjpv/edenBooks/internal/modules/chats/application/dto"
	"github.com/javierjpv/edenBooks/internal/modules/chats/domain/entities"
)

type ChatService interface {
	CreateChat(chat dto.ChatRequest) (*entities.Chat, error)

	CreateChatWithUsers(userIDs []uint) (*entities.Chat, error)

	UpdateChat(id uint, chat dto.ChatRequest) error

	DeleteChat(id uint) error

	GetChatByID(id uint) (*entities.Chat, error)

	GetFilteredChats(filters map[string]string) ([]entities.Chat, error)
}

File: frontend/edenBooks/src/modules/chats/services/chatService.ts
import { IChatRequest } from "../interfaces/IChatRequest";
import { IChatService } from "../interfaces/IChatService";
import { chatApi } from "../api/chatApi";


export const chatService: IChatService = {
  CreateChat: async (chat: IChatRequest) =>chatApi.CreateChat(chat),
  GetChats: async () => chatApi.GetChats(),
  GetChatById: async (id: number) => chatApi.GetChatById(id),
  UpdateChat: async (id: number, chat: IChatRequest) =>chatApi.UpdateChat(id, chat),
  DeleteChat: async (id: number) => chatApi.DeleteChat(id),
};
